### PR TITLE
feat!: New Tabs

### PR DESCRIPTION
**Pull request by jakemhiller**
### Overview

Replaces deprecated `Tabs` component with [reach-ui/tabs](https://reach.tech/tabs/) based tabs components

Issues found during migration:

- [ ]  system props not being allowed by types on `TabsList`
- [ ] very few components seem to allow `className`

### PR Checklist

- [ ] Related to designs: https://www.figma.com/file/RfVSt2smXRygdjHYSDJAMa/Navigation-Pattern-Cleanup?node-id=108%3A463
- [ ] Related to JIRA ticket: [GM-443]
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### TODO

- [ ] Tests (they will be minimal, since we're not overriding reach-ui/tabs functionality, which is well tested)
- [ ] Documentation, changes I did make need clean-up and overall stories need more detail

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.19-alpha.63f087.0<br/>@codecademy/gamut-labs@28.0.14-alpha.63f087.0<br/>@codecademy/gamut@40.3.4-alpha.63f087.0<br/>@codecademy/monaco@0.5.14-alpha.63f087.0<br/>@codecademy/styleguide@50.2.6-alpha.63f087.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6197ffbe42e2801ea807d156--gamut-preview.netlify.app>https://6197ffbe42e2801ea807d156--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6197ffbe42e2801ea807d156>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.19-alpha.208b84.0<br/>@codecademy/gamut-labs@28.0.14-alpha.208b84.0<br/>@codecademy/gamut@40.3.4-alpha.208b84.0<br/>@codecademy/monaco@0.5.14-alpha.208b84.0<br/>@codecademy/styleguide@50.2.6-alpha.208b84.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.19-alpha.34d2bd.0<br/>@codecademy/gamut-labs@28.0.14-alpha.34d2bd.0<br/>@codecademy/gamut@40.3.4-alpha.34d2bd.0<br/>@codecademy/monaco@0.5.14-alpha.34d2bd.0<br/>@codecademy/styleguide@50.2.6-alpha.34d2bd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://619815e8f8e697346ee4fad7--gamut-preview.netlify.app>https://619815e8f8e697346ee4fad7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/619815e8f8e697346ee4fad7>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.19-alpha.b3578c.0<br/>@codecademy/gamut-labs@28.0.14-alpha.b3578c.0<br/>@codecademy/gamut@40.3.4-alpha.b3578c.0<br/>@codecademy/monaco@0.5.14-alpha.b3578c.0<br/>@codecademy/styleguide@50.2.7-alpha.b3578c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61983cb287d83f302e0832c3--gamut-preview.netlify.app>https://61983cb287d83f302e0832c3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61983cb287d83f302e0832c3>Deploy Logs</a></p>

File: packages/gamut/src/Tabs/Tabs/index.tsx [hunk 1]
```diff
@@ -1,112 +1,24 @@
...
-export interface TabsState {
-  activeTabIndex: number;
+import { css, system } from '@codecademy/gamut-styles';
+import { StyleProps, variance } from '@codecademy/variance';
+import styled from '@emotion/styled';
+import { Tabs as ReachTabs, TabsProps as ReachTabsProps } from '@reach/tabs';
+
+// Prevent dev-only errors due to excluding react-ui default styles
+if (process.env.NODE_ENV !== 'production' && document?.documentElement) {
+  document.documentElement.style.setProperty('--reach-tabs', '1');
```
Comment (dreamwasp) [inline packages/gamut/src/Tabs/Tabs/index.tsx:8]: nice

**Review by dreamwasp — COMMENTED** [review]
this looks great + def makes `TabsExperimental` very killable. only thing that stood out is that we still have a blue focus state when tabbing around 

image

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.19-alpha.848650.0<br/>@codecademy/gamut-labs@28.0.14-alpha.848650.0<br/>@codecademy/gamut@40.3.4-alpha.848650.0<br/>@codecademy/monaco@0.5.14-alpha.848650.0<br/>@codecademy/styleguide@50.2.7-alpha.848650.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.21-alpha.434238.0<br/>@codecademy/gamut-labs@28.1.1-alpha.434238.0<br/>@codecademy/gamut@40.4.2-alpha.434238.0<br/>@codecademy/monaco@0.5.16-alpha.434238.0<br/>@codecademy/styleguide@50.4.1-alpha.434238.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://619e74ad853a7c8b73b82ffc--gamut-preview.netlify.app>https://619e74ad853a7c8b73b82ffc--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/619e74ad853a7c8b73b82ffc>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.21-alpha.0ff0b6.0<br/>@codecademy/gamut-labs@28.1.1-alpha.0ff0b6.0<br/>@codecademy/gamut@40.4.2-alpha.0ff0b6.0<br/>@codecademy/monaco@0.5.16-alpha.0ff0b6.0<br/>@codecademy/styleguide@50.4.1-alpha.0ff0b6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://619eb46965d1eed6c90584d9--gamut-preview.netlify.app>https://619eb46965d1eed6c90584d9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/619eb46965d1eed6c90584d9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.21-alpha.13dc90.0<br/>@codecademy/gamut-labs@28.1.1-alpha.13dc90.0<br/>@codecademy/gamut@40.4.2-alpha.13dc90.0<br/>@codecademy/monaco@0.5.16-alpha.13dc90.0<br/>@codecademy/styleguide@50.4.1-alpha.13dc90.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://619eb669d18e14d8138e63a0--gamut-preview.netlify.app>https://619eb669d18e14d8138e63a0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/619eb669d18e14d8138e63a0>Deploy Logs</a></p>

**Comment by jakemhiller** [comment]
> this looks great + def makes `TabsExperimental` very killable. only thing that stood out is that we still have a blue focus state when tabbing around
> 
> image

This should be fixed

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.22-alpha.3f6441.0<br/>@codecademy/gamut-labs@28.1.2-alpha.3f6441.0<br/>@codecademy/gamut@40.5.1-alpha.3f6441.0<br/>@codecademy/monaco@0.5.17-alpha.3f6441.0<br/>@codecademy/styleguide@50.4.2-alpha.3f6441.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61a6452686b8332eb811b8d5--gamut-preview.netlify.app>https://61a6452686b8332eb811b8d5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61a6452686b8332eb811b8d5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.22-alpha.995c67.0<br/>@codecademy/gamut-labs@28.1.2-alpha.995c67.0<br/>@codecademy/gamut@40.5.1-alpha.995c67.0<br/>@codecademy/monaco@0.5.17-alpha.995c67.0<br/>@codecademy/styleguide@50.4.2-alpha.995c67.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61a64daa01d31435a99f14bf--gamut-preview.netlify.app>https://61a64daa01d31435a99f14bf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61a64daa01d31435a99f14bf>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.22-alpha.2a3b74.0<br/>@codecademy/gamut-labs@28.1.2-alpha.2a3b74.0<br/>@codecademy/gamut@40.5.1-alpha.2a3b74.0<br/>@codecademy/monaco@0.5.17-alpha.2a3b74.0<br/>@codecademy/styleguide@50.4.2-alpha.2a3b74.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61a92a6787ed057eb308b56f--gamut-preview.netlify.app>https://61a92a6787ed057eb308b56f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61a92a6787ed057eb308b56f>Deploy Logs</a></p>

**Comment by thedoomshine** [comment]
love the accessibility and keyboard navigation! 🔥

**Review by vonbarown — APPROVED** [review]
LGTM

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.22-alpha.4350fa.0<br/>@codecademy/gamut-labs@28.1.2-alpha.4350fa.0<br/>@codecademy/gamut@40.5.1-alpha.4350fa.0<br/>@codecademy/monaco@0.5.17-alpha.4350fa.0<br/>@codecademy/styleguide@50.4.2-alpha.4350fa.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61b0f0244920e11e7e469bf3--gamut-preview.netlify.app>https://61b0f0244920e11e7e469bf3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61b0f0244920e11e7e469bf3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.22-alpha.03d5c4.0<br/>@codecademy/gamut-labs@28.1.2-alpha.03d5c4.0<br/>@codecademy/gamut@40.5.1-alpha.03d5c4.0<br/>@codecademy/monaco@0.5.17-alpha.03d5c4.0<br/>@codecademy/styleguide@50.4.2-alpha.03d5c4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61b1238b9d8b825289fec677--gamut-preview.netlify.app>https://61b1238b9d8b825289fec677--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61b1238b9d8b825289fec677>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.22-alpha.f6775a.0<br/>@codecademy/gamut-labs@28.1.2-alpha.f6775a.0<br/>@codecademy/gamut@40.5.1-alpha.f6775a.0<br/>@codecademy/monaco@0.5.17-alpha.f6775a.0<br/>@codecademy/styleguide@50.4.2-alpha.f6775a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61b1259a7c3a0a5b1cdcf8c4--gamut-preview.netlify.app>https://61b1259a7c3a0a5b1cdcf8c4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61b1259a7c3a0a5b1cdcf8c4>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
yes! looks (+ sounds!) great  to me, just wanna make sure it still works in where its used in the [monolith](https://github.com/codecademy-engineering/Codecademy/blob/develop/webpack/assets/platform/scenes/kanbanProject/kanbanProjectContainer/kanbanProjectSidebar/index.tsx).

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.28-alpha.a6b263.0<br/>@codecademy/gamut-labs@28.3.2-alpha.a6b263.0<br/>@codecademy/gamut@41.0.1-alpha.a6b263.0<br/>@codecademy/monaco@0.5.21-alpha.a6b263.0<br/>@codecademy/styleguide@51.0.1-alpha.a6b263.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61b3901662aefd8aeaa22f35--gamut-preview.netlify.app>https://61b3901662aefd8aeaa22f35--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61b3901662aefd8aeaa22f35>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.28-alpha.620462.0<br/>@codecademy/gamut-labs@28.3.2-alpha.620462.0<br/>@codecademy/gamut@41.0.1-alpha.620462.0<br/>@codecademy/monaco@0.5.21-alpha.620462.0<br/>@codecademy/styleguide@51.0.1-alpha.620462.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61b393e0c2eeb3813c14a6ff--gamut-preview.netlify.app>https://61b393e0c2eeb3813c14a6ff--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61b393e0c2eeb3813c14a6ff>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.28-alpha.674394.0<br/>@codecademy/gamut-labs@28.3.2-alpha.674394.0<br/>@codecademy/gamut@41.0.1-alpha.674394.0<br/>@codecademy/monaco@0.5.21-alpha.674394.0<br/>@codecademy/styleguide@51.0.1-alpha.674394.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61b39796680b425d09aaba8e--gamut-preview.netlify.app>https://61b39796680b425d09aaba8e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61b39796680b425d09aaba8e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.29-alpha.39766f.0<br/>@codecademy/gamut-labs@28.3.3-alpha.39766f.0<br/>@codecademy/gamut@41.1.1-alpha.39766f.0<br/>@codecademy/monaco@0.5.22-alpha.39766f.0<br/>@codecademy/styleguide@51.1.1-alpha.39766f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61b3995063cfc091efc7a3cf--gamut-preview.netlify.app>https://61b3995063cfc091efc7a3cf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61b3995063cfc091efc7a3cf>Deploy Logs</a></p>

**Pull request merged**